Modules/posixmodule.c

```diff
@@ -13996,6 +13996,82 @@ os_waitstatus_to_exitcode_impl(PyObject *module, PyObject *status_obj)
 }
 #endif
 
+#ifdef MS_WINDOWS
+static PyObject *
+windows_namespace(void)
+{
+    PyObject *d = PyDict_New(), *ret = NULL, *consts = NULL, *errors = NULL;
+    if (d == NULL) {
+        return NULL;
+    }
+
+#define SET(name) \
+    { \
+    PyObject *o = PyLong_FromLong(name); \
+    if (o == NULL) { \
+        goto error; \
+    } \
+    int res = PyDict_SetItemString(d, #name, o); \
+    Py_DECREF(o); \
+    if (res < 0) { \
+        goto error; \
+    } \
+    } \
+
+    SET(DELETE);
+    SET(READ_CONTROL);
+    SET(WRITE_DAC);
+    SET(WRITE_OWNER);
+    SET(SYNCHRONIZE);
+    SET(STANDARD_RIGHTS_REQUIRED);
+    SET(STANDARD_RIGHTS_READ);
+    SET(STANDARD_RIGHTS_WRITE);
+    SET(STANDARD_RIGHTS_EXECUTE);
+    SET(STANDARD_RIGHTS_ALL);
+    SET(SPECIFIC_RIGHTS_ALL);
+    consts = _PyNamespace_New(d);
+    if (consts == NULL) {
+        goto error;
+    }
+    PyDict_Clear(d);
+
+    SET(ERROR_ALREADY_EXISTS);
+    SET(ERROR_BROKEN_PIPE);
+    SET(ERROR_IO_PENDING);
+    SET(ERROR_MORE_DATA);
+    SET(ERROR_NETNAME_DELETED);
+    SET(ERROR_NO_SYSTEM_RESOURCES);
+    SET(ERROR_MORE_DATA);
+    SET(ERROR_NETNAME_DELETED);
+    SET(ERROR_NO_DATA);
+    SET(ERROR_NO_SYSTEM_RESOURCES);
+    SET(ERROR_OPERATION_ABORTED);
+    SET(ERROR_PIPE_BUSY);
+    SET(ERROR_PIPE_CONNECTED);
+    SET(ERROR_SEM_TIMEOUT);
+#undef SET
```

Comment:
The following error codes are listed twice: `ERROR_MORE_DATA`, `ERROR_NETNAME_DELETED`, and `ERROR_NO_SYSTEM_RESOURCES`.

I'd also add the following common codes. Most of these have an explicit `errno` mapping, but usually several codes map to the same `errno` value.

```C
SET(ERROR_GEN_FAILURE);
SET(ERROR_INVALID_FUNCTION);
SET(ERROR_NOT_SUPPORTED);
SET(ERROR_INVALID_PARAMETER);
SET(ERROR_INVALID_HANDLE);
SET(ERROR_INVALID_NAME);
SET(ERROR_BAD_NET_NAME);
SET(ERROR_BAD_PATHNAME);
SET(ERROR_FILENAME_EXCED_RANGE);
SET(ERROR_FILE_NOT_FOUND);
SET(ERROR_PATH_NOT_FOUND);
SET(ERROR_BAD_NETPATH);
SET(ERROR_NO_MORE_FILES);
SET(ERROR_FILE_EXISTS);
SET(ERROR_NOT_SAME_DEVICE);
SET(ERROR_DIRECTORY);
SET(ERROR_TOO_MANY_OPEN_FILES);
SET(ERROR_DISK_FULL);
SET(ERROR_DIR_NOT_EMPTY);
SET(ERROR_CANT_ACCESS_FILE);
SET(ERROR_CANT_RESOLVE_FILENAME);
SET(ERROR_NOT_A_REPARSE_POINT);
SET(ERROR_INVALID_REPARSE_DATA);
SET(ERROR_REPARSE_TAG_INVALID);
SET(ERROR_ACCESS_DENIED);
SET(ERROR_NETWORK_ACCESS_DENIED);
SET(ERROR_SHARING_VIOLATION);
SET(ERROR_LOCK_VIOLATION);
SET(ERROR_NOT_READY);
SET(ERROR_NOT_LOCKED);
SET(ERROR_LOCK_FAILED);
SET(ERROR_BAD_EXE_FORMAT);
SET(ERROR_NOT_ENOUGH_MEMORY);
SET(ERROR_NOT_ENOUGH_QUOTA);
```

Reply:
Here are few more error codes to consider:

```C
SET(ERROR_BAD_PIPE);
SET(ERROR_PIPE_NOT_CONNECTED);
SET(ERROR_PIPE_LISTENING);
SET(ERROR_ELEVATION_REQUIRED);
SET(ERROR_PRIVILEGE_NOT_HELD);
SET(ERROR_SYMLINK_CLASS_DISABLED);
```

* `ERROR_BAD_PIPE` is from NT `STATUS_INVALID_PIPE_STATE` or  `STATUS_INVALID_READ_MODE`.
* `ERROR_ELEVATION_REQUIRED` is returned by `CreateProcessW` if an executable requires elevation (e.g. regedit.exe or osk.exe). It can be handled by calling `ShellExecute[Ex]W` with the "runas" verb. 
* `ERROR_PRIVILEGE_NOT_HELD` means the caller lacks a required privilege, or in some cases that the privilege isn't enabled. 
* `ERROR_SYMLINK_CLASS_DISABLED` means the system's L2L (local to local) symlink policy has disabled evaluation of local symlinks that target local paths. If the L2R (local to remote), R2L, or R2R policy is disabled,  `ERROR_NETWORK_ACCESS_DENIED` is set instead. The link can be resolved manually via `os.readlink`, but doing so violates system policy.

For checking exit status, here are some abnormal-termination status codes. This includes process initialization failures, unhandled exceptions, and unhandled console control events (Ctrl+C, Ctrl+Break, Ctrl+Close). Other than for testing purposes or providing a friendlier error message, usually there's nothing to handle in these cases.

```C
SET(STATUS_ENTRYPOINT_NOT_FOUND);
SET(STATUS_DLL_NOT_FOUND);
SET(STATUS_ORDINAL_NOT_FOUND);
SET(STATUS_DLL_INIT_FAILED);
SET(STATUS_ACCESS_VIOLATION);
SET(STATUS_HEAP_CORRUPTION);
SET(STATUS_STACK_BUFFER_OVERRUN);
SET(STATUS_STACK_OVERFLOW);
SET(STATUS_ILLEGAL_INSTRUCTION);
SET(STATUS_PRIVILEGED_INSTRUCTION);
SET(STATUS_CONTROL_C_EXIT);
```

`STATUS_STACK_BUFFER_OVERRUN` (0xC000_0409) is also used by the fastfail intrinsic, e.g. `os.abort()`. The reason for the fastfail is included in the exception record and the corresponding Windows Error Reporting event in the application event log -- e.g. `FAST_FAIL_FATAL_APP_EXIT` (7) for an abort.

If the process is still running, the status is `STATUS_PENDING`. In the Windows API it's the same value aliased as `STILL_ACTIVE`:

```C
SET(STATUS_PENDING);
SET(STILL_ACTIVE);
```